google/cloud/storage/examples/storage_object_samples.cc

```diff
@@ -178,6 +178,44 @@ void InsertObjectModifiedRetry(google::cloud::storage::Client unused, int& argc,
   (bucket_name, object_name, contents);
 }
 
+void InsertObjectMultipart(google::cloud::storage::Client client, int& argc,
+                           char* argv[]) {
+  if (argc != 5) {
+    throw Usage{
+        "insert-object-multipart <bucket-name> <object-name>"
+        " <content-type> <object-contents (string)>"};
+  }
+  auto bucket_name = ConsumeArg(argc, argv);
+  auto object_name = ConsumeArg(argc, argv);
+  auto content_type = ConsumeArg(argc, argv);
+  auto contents = ConsumeArg(argc, argv);
+  //! [insert object multipart]
+  namespace gcs = google::cloud::storage;
+  using google::cloud::StatusOr;
+  [](gcs::Client client, std::string bucket_name, std::string object_name,
+     std::string content_type, std::string contents) {
+    StatusOr<gcs::ObjectMetadata> object_metadata = client.InsertObject(
+        bucket_name, object_name, std::move(contents),
+        gcs::WithObjectMetadata(
+            gcs::ObjectMetadata().set_content_type(content_type)));
```

Comment:
Suggestion: IIUC, this is only multipart when `gcs::WithObjectMetadata()` is provided and otherwise it's a simple upload. It would help to add a comment above the request, because it's subtle.

Reply:
Yup, fixed.